internal/fs/fs.go

```diff
@@ -2252,26 +2258,29 @@ func (fs *fileSystem) Unlink(
 	fs.mu.Lock()
 	parent := fs.dirInodeOrDie(op.Parent)
 	fs.mu.Unlock()
-
-	// if inode is a local file, mark it unlinked.
 	fileName := inode.NewFileName(parent.Name(), op.Name)
+
+	//If the inode represents a local file, mark it as unlinked and return.
 	fs.mu.Lock()
-	fileInode, ok := fs.localFileInodes[fileName]
```

Comment:
I am trying to avoid multiple lock, unlock flow. Can we do something like this:
fs.mu.lock
inode, ok = fs.localFileInodes[fileName]
if !ok {
   inode, ok = fs.fs.generationBackedInodes[fileName]
}
fs.mu.unlock

if inode != nil {
  inode.unlink
}